osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/metrics/PrometheusMetricsServer.java

```diff
@@ -67,7 +70,7 @@
   @Value("${metrics.prometheus.path}")
   private String path;
 
-  @Value("${metrics.componentname}")
+  @Value("${metrics.componentname:}")
```

Comment:
Looks a bit strange. If the intention is to make the component name optional (not sure if that is a good choice by itself) it looks odd to call `requireNonNull(this.componentName, "...")` in the `@PostConstruct` method `start`.
Or is there some particular reason that a blank String is OK, but `null` is not?

Reply:
My objection was not to making the component name optional. If this makes sense now that is OK with me.
What bothers me is, that if the component name has become optional, the following check does not seem to make sense:
```java
requireNonNull(this.componentName, "Component name not set");
```
It would appear to be more logical to leave that requirement not to be null out, and have null values treated the same as an empty String in: `if (!StringUtils.isEmpty(this.componentName)) {`.
Or is there something I am missing that would make it logical to get an error on null, demanding the component name to be set, while it has become an optional thing?